.github/workflows/array_api.yml

```diff
@@ -34,76 +32,31 @@ jobs:
       needs.get_commit_message.outputs.message == 1
       && (github.repository == 'scipy/scipy' || github.repository == '')
     runs-on: ubuntu-22.04
-    strategy:
-      matrix:
-        python-version: ["3.12"]
-
     steps:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           submodules: recursive
 
-      - name: Setup Python
-        uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
+      - uses: prefix-dev/setup-pixi@ba3bb36eb2066252b2363392b7739741bb777659 # v0.8.1
         with:
-          python-version: ${{ matrix.python-version }}
-          cache: "pip" # not using a path to also cache pytorch
-
-      - name: Install Ubuntu dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y libopenblas-dev libatlas-base-dev liblapack-dev gfortran libgmp-dev libmpfr-dev libsuitesparse-dev ccache libmpc-dev
-
-      - name: Install Python packages
-        run: |
-          python -m pip install numpy cython pytest pytest-xdist pytest-timeout pybind11 mpmath gmpy2 pythran ninja meson pooch hypothesis array-api-strict spin "click<8.3.0"
-
-      - name: Install PyTorch CPU
-        run: |
-          python -m pip install torch --index-url https://download.pytorch.org/whl/cpu
-
-      - name: Install JAX
-        run: |
-          # Skip 0.6.2: https://github.com/jax-ml/jax/issues/29537
-          python -m pip install "jax[cpu]!=0.6.2"
-
-      - name: Install Dask
-        run: |
-          python -m pip install dask[array]
-
-      - name: Install MArray
-        run: |
-          python -m pip install marray
-
-      - name: Prepare compiler cache
-        id: prep-ccache
-        shell: bash
-        run: |
-          mkdir -p "${CCACHE_DIR}"
-          echo "dir=$CCACHE_DIR" >> $GITHUB_OUTPUT
-          NOW=$(date -u +"%F-%T")
-          echo "timestamp=${NOW}" >> $GITHUB_OUTPUT
-
-      - name: Setup compiler cache
-        uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
-        id: cache-ccache
+          pixi-version: v0.55.0
+          manifest-path: .github/workflows/pixi.toml
+          cache-write: ${{ github.event_name == 'push' && github.ref_name == 'main' }}
+          environments: >-
+            build
+            array-api-cpu
+
+      - name: Set up ccache
+        uses: ./.github/ccache
         with:
-          path: ${{ steps.prep-ccache.outputs.dir }}
-          key: ${{ github.workflow }}-${{ matrix.python-version }}-ccache-linux-${{ steps.prep-ccache.outputs.timestamp }}
-          restore-keys: |
-            ${{ github.workflow }}-${{ matrix.python-version }}-ccache-linux-
-
-      - name: Setup build and install scipy
```

Comment:
I'd prefer for building to stay a separate step - it makes it much easier to find important data for the build and test steps (e.g., the discovery output `pytest` prints out). And it's shorter logs per step, making navigation in the GitHub UI faster.